### ISSUE TITLE
List scroll API enhancements

### ISSUE THREAD

**Issue by dweymouth** [fyne-io/fyne#3544]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

I would like to be able to get the current scroll position of a list, and programmatically scroll a list to a specified position (not list item ID). In my case this would be useful because I would like to be able to save just the data about a list (the backing model plus the current scroll position) so I can rebuild the list widget later exactly as it appeared before saving it.

### Is it possible to construct a solution with the existing API?

Not really, `ScrollTo` exists to scroll to a specific item, but there is no API to get the current scroll position. `ScrollTo` is also limited since it can't scroll to an arbitrary position (i.e. in between two items)

### Describe the solution you'd like to see.

Related to #2105, I'd like to see further extensions to the list scrolling API to enable getting the current scroll position, and scrolling to a specified position. Something like

```
func (l *List) GetScrollPosition() fyne.Position
func (l *List) ScrollToPosition(pos fyne.Position)
```

And maybe also something like:

```
func (l *List) GetFirstVisibleItemID() list.ListItemID
```

**Comment by dweymouth** [comment]
This already exists in GridWrap (and I use it in Supersonic), so I'm pulling it into the 2.5 milestone to add it for List as well.

**Comment by andydotxyz** [comment]
Looks like this was resolved on develop for 2.5 :)
